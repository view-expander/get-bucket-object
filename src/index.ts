import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'
import ImgixClient from 'imgix-core-js'

export async function handler(
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> {
  try {
    const requestBody = event.body === null ? {} : JSON.parse(event.body)
    const key = requestBody.key

    if (typeof key !== 'string') {
      throw new Error('Object key is required')
    }

    if (process.env.IMGIX_DOMAIN === undefined) {
      throw new Error('the domain of imgix is required')
    }

    const size = event.queryStringParameters?.size
    const client = new ImgixClient({ domain: process.env.IMGIX_DOMAIN })
    const url = client.buildURL(
      key,
      size === 'thumb'
        ? {
            fit: 'clip',
            h: 256,
            w: 256,
          }
        : undefined
    )

    return {
      statusCode: 301,
      headers: {
        Location: url,
      },
      body: JSON.stringify({ path: url }),
    }
  } catch (err) {
    return {
      statusCode: err.statusCode || 400,
      body: JSON.stringify(err.message),
    }
  }
}
