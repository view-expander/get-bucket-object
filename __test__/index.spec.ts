import { APIGatewayProxyEvent } from 'aws-lambda'
import { handler } from '../src'

describe('handler()', () => {
  it('returns dummy response', async () => {
    expect.assertions(1)

    return expect(handler({} as APIGatewayProxyEvent)).resolves.toEqual({
      statusCode: 200,
      body: '',
    })
  })
})
