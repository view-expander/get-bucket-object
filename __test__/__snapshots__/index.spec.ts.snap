// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler() be succeed 1`] = `
Object {
  "body": "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQI12NgYAAAAAMAASDVlMcAAAAASUVORK5CYII=",
  "headers": Object {
    "Content-Type": "image/png",
  },
  "isBase64Encoded": true,
  "statusCode": 200,
}
`;
